PyInstaller/building/api.py

```diff
@@ -532,7 +532,8 @@ def assemble(self):
             raise SystemExit(_MISSING_BOOTLOADER_ERRORMSG)
 
         if is_win and (self.icon != "NONE" or self.versrsrc or self.resources
-                       or self.uac_admin or self.uac_uiaccess or not is_64bits):
+                       or self.uac_admin or self.uac_uiaccess or not is_64bits
+                       or (self.manifest and not self.exclude_binaries)):
```

Comment:
I'd rather like some comments in here explaining why simply `if is_win:` is not what we want. Is it that there's no point adding a manifest unless at least one of those conditions is true or that it breaks it if we add the manifest anyway?

Reply:
Yeah, that's a fair point. I think going with just `is_win` should be enough (and I strongly suspect that this part of code simply gradually evolved to be that way).

The optional stuff (`self.icon`, `self.versrsrc` and `self.resources`) are under their corresponding subsequent checks anyway. 

The (self.manifest and not self.exclude_binaries) was added to fix this issue (and its subsequent check has an additional `is_win` check that's redundant).

`self.uac_admin` or `self.uac_uiaccess` are probably there to ensure that we add the manifest if either of those options is enabled (as they're written in the manifest).

Not sure about `not is_64bits` part, but I suspect that nowdays the `self.icon != "NONE"` accounts for most cases, so that doesn't influence much, anyway.

Reply:
> Is it that there's no point adding a manifest unless at least one of those conditions is true or that it breaks it if we unconditionally add the manifest ?

We probably always need the manifest. If nothing else, we use it to try opting-in for long paths; and as discovered by #5624, some other things may depend on it as well.

The only time when we don't need to embed it in the executable is if we're doing `onedir` build, since it's written to a separate file. (Which I think is why we have that `not self.exclude_binaries` check).

Reply:
Tracing the blame back, it does look like it started of as *add if needed* and `*if needed* grew and grew:
- f11fb8b87f8797b7e086a09f13be94003554b777
- 3c3228da89c494f79dc1cead99ab9b4e45a7424e
- 1905d2bc1e33cadc40f8dbe648aad3452b9b82fd

>  (Which I think is why we have that not self.exclude_binaries check).

Hmm, possibly. It looks like there isn't just a `self.is_onefile` flag.